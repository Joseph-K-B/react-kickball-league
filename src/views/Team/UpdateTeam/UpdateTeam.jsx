import { useEffect, useState } from "react";
import { useHistory, useParams } from "react-router-dom";
import { Link } from "react-router-dom";
import TeamForm from "../../../components/TeamForm";
import { getTeamById, updateTeamById } from "../../../services/teams";


function UpdateTeam() {
    const { teamId } = useParams();
    const [team, setTeam] = useState(null);
    const [name, setName] = useState('');
    const [city, setCity] = useState('');
    const [state, setState] = useState('');
    const history = useHistory();

    useEffect(() => {
        getTeamById(teamId).then((res) => setTeam(res))
    }, [teamId]);

    const handleUpdateSubmit = async (e) => {
        e.preventDefault();
        const res = await updateTeamById( teamId, {name, city, state})
        history.push(`/teams/${res[0].id}`)
    };

    if(!team) return <h2>Loading Team...</h2>

    return (
        <>
        <h1>{team.name}</h1>
        <h3>{team.city}, {team.state}</h3>
        <p>Wins: {team.wins}</p>
        <ul>
        {team.players.map((player) => (
            <li key={player.id}>
                <Link className='link' to={`/players/${player.id}`}>
                    <p>{player.position} - {player.name}</p>
                </Link>
            </li>
        ))}
        </ul>
        <fieldset>
            <legend>Update Team</legend>
            <TeamForm 
                name={name}
                city={city}
                state={state}
                handleSubmit={handleUpdateSubmit}
                setName={setName}
                setCity={setCity}
                setState={setState}
            />
        </fieldset>
        <Link className='link' to='/teams'>
            <p>Team Directory</p>
        </Link>        
        </>
    )
}

export default UpdateTeam;