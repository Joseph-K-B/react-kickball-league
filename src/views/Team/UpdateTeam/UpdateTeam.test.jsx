import { render, screen } from '@testing-library/react';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import {Router, Route } from 'react-router-dom';
import { createMemoryHistory } from "history";
import userEvent from '@testing-library/user-event';
import UpdateTeam from './UpdateTeam'

const mockTeam = {
    id: 20,
    created_at: '2021-12-08T16:09:33.408898+00:00',
    name: 'name',
    city: 'city',
    state: 'state',
    players: []
}

const server = setupServer(
    rest.get('https://vrmauzdnhcbzknrntyjm.supabase.co/rest/v1/teams', 
    (req, res, ctx) => {
        return res(ctx.json(mockTeam));
    }),
    rest.post('https://vrmauzdnhcbzknrntyjm.supabase.co/rest/v1/teams', 
    (req, res, ctx) => {
        return res (
            ctx.json([mockTeam]));
    })
);

beforeAll(() => {
    server.listen();
});

afterAll(() => {
    server.close();
});

it('renders form to update team', async () => {
    const history = createMemoryHistory();
    history.push('/teams/1/update');

    const {container} = render(
        <Router history={history}>
            <Route>
                <UpdateTeam />
            </Route>
        </Router>
    );

    await screen.findByText('Update Team');

    const nameField = screen.getByLabelText(/Name:/);
    const cityField = screen.getByLabelText(/City:/);
    const stateField = screen.getByLabelText(/State:/);
    const submitBtn = screen.getByRole('button', {name: 'edit team' });

    userEvent.type(nameField, 'Another Team');
    userEvent.type(cityField, 'Another Town');
    userEvent.type(stateField, 'US');
    userEvent.click(submitBtn);

    expect(container).toMatchSnapshot();
    await screen.findByText('name');
});