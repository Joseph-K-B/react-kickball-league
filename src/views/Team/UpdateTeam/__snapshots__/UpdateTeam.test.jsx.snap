// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders form to update team 1`] = `
<div>
  <h1>
    name
  </h1>
  <h3>
    city
    , 
    state
  </h3>
  <p>
    Wins: 
  </p>
  <ul />
  <fieldset>
    <legend>
      Update Team
    </legend>
    <form>
      <label
        for="name"
      >
        Name:
      </label>
      <input
        id="name"
        type="text"
        value="Another Team"
      />
      <label
        for="city"
      >
        City:
      </label>
      <input
        id="city"
        type="text"
        value="Another Town"
      />
      <label
        for="state"
      >
        State:
      </label>
      <input
        id="state"
        type="text"
        value="US"
      />
      <button
        aria-label="edit team"
        type="submit"
      >
        Submit
      </button>
    </form>
  </fieldset>
  <a
    class="link"
    href="/teams"
  >
    <p>
      Team Directory
    </p>
  </a>
</div>
`;
